pkg/jsonschema/generate_config_docs.go

```diff
@@ -0,0 +1,270 @@
+		lo.Map(strings.Split(description, "\n"), func(s string, _ int) string {
+			if s == "" {
+				return "#" // avoid trailing space on blank lines
+			}
+			return "# " + s
+		}),
+		"\n")
+}
+
+func (n *Node) MarshalYAML() (interface{}, error) {
+	node := yaml.Node{
+		Kind: yaml.MappingNode,
+	}
+
+	keyNode := yaml.Node{
+		Kind:  yaml.ScalarNode,
+		Value: n.Name,
+	}
+	if n.Description != "" {
+		setComment(&keyNode, n.Description)
+	}
+
+	if n.Default != nil {
+		valueNode := yaml.Node{
+			Kind: yaml.ScalarNode,
+		}
+		err := valueNode.Encode(n.Default)
+		if err != nil {
+			return nil, err
+		}
+		node.Content = append(node.Content, &keyNode, &valueNode)
+	} else if len(n.Children) > 0 {
+		childrenNode := yaml.Node{
+			Kind: yaml.MappingNode,
+		}
+		for _, child := range n.Children {
+			childYaml, err := child.MarshalYAML()
+			if err != nil {
+				return nil, err
+			}
+
+			childKey := yaml.Node{
+				Kind:  yaml.ScalarNode,
+				Value: child.Name,
+			}
+			if child.Description != "" {
+				setComment(&childKey, child.Description)
+			}
+			childYaml = childYaml.(*yaml.Node)
+			childrenNode.Content = append(childrenNode.Content, childYaml.(*yaml.Node).Content...)
+		}
+		node.Content = append(node.Content, &keyNode, &childrenNode)
+	}
+
+	return &node, nil
+}
+
+func getDescription(v *orderedmap.OrderedMap) string {
+	description, ok := v.Get("description")
+	if !ok {
+		description = ""
+	}
+	return description.(string)
+}
+
+func getDefault(v *orderedmap.OrderedMap) (error, any) {
+	defaultValue, ok := v.Get("default")
+	if ok {
+		return nil, defaultValue
+	}
+
+	dataType, ok := v.Get("type")
+	if ok {
+		dataTypeString := dataType.(string)
+		if dataTypeString == "string" {
+			return nil, ""
+		}
+	}
+
+	return fmt.Errorf("Failed to get default value"), nil
+}
+
+func parseNode(parent *Node, name string, value *orderedmap.OrderedMap) {
+	description := getDescription(value)
+	err, defaultValue := getDefault(value)
+	if err == nil {
+		leaf := &Node{Name: name, Description: description, Default: defaultValue}
+		parent.Children = append(parent.Children, leaf)
+	}
+
+	properties, ok := value.Get("properties")
+	if !ok {
+		return
+	}
+
+	orderedProperties := properties.(orderedmap.OrderedMap)
+
+	node := &Node{Name: name, Description: description}
+	parent.Children = append(parent.Children, node)
+
+	keys := orderedProperties.Keys()
+	for _, name := range keys {
+		value, _ := orderedProperties.Get(name)
+		typedValue := value.(orderedmap.OrderedMap)
+		parseNode(node, name, &typedValue)
+	}
+}
+
+func writeToConfigDocs(config []byte) error {
+	configPath := utils.GetLazyRootDirectory() + "/docs/Config.md"
+	markdown, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("Error reading Config.md file %w", err)
+	}
+
+	startConfigSectionIndex := bytes.Index(markdown, []byte(DocumentationCommentStart))
+	if startConfigSectionIndex == -1 {
+		return fmt.Errorf("Default config starting comment not found, %w", err)
```

Comment:
`err` is always nil here, so it doesn't make sense to include it in the error message. Also, when no formatting is needed, it's better to use `errors.New` instead of `fmt.Errorf` (see #3523). Changed in dd9f1581748.